runtime/fs_node.js

```diff
@@ -104,17 +126,125 @@ MlNodeDevice.prototype.open = function(name, f) {
     var fd = this.fs.openSync(this.nm(name), res);
     return new MlNodeFile(fd);
   } catch (err) {
-    caml_raise_sys_error(err.toString());
+    this.raise_nodejs_error(err, raise_unix);
   }
 }
 
-MlNodeDevice.prototype.rename = function(o,n) {
+MlNodeDevice.prototype.rename = function(o, n, raise_unix) {
   try {
     this.fs.renameSync(this.nm(o), this.nm(n));
   } catch (err) {
+    this.raise_nodejs_error(err, raise_unix);
+  }
+}
+MlNodeDevice.prototype.stat = function(name, raise_unix) {
+  try {
+    var js_stats = this.fs.statSync(this.nm(name));
+    return this.stats_from_js(js_stats);
+  } catch (err) {
+    this.raise_nodejs_error(err, raise_unix);
+  }
+}
+MlNodeDevice.prototype.lstat = function(name, raise_unix) {
+  try {
+    var js_stats = this.fs.lstatSync(this.nm(name));
+    return this.stats_from_js(js_stats);
+  } catch (err) {
+    this.raise_nodejs_error(err, raise_unix);
+  }
+}
+MlNodeDevice.prototype.symlink = function(to_dir, src, dst, raise_unix) {
+  try {
+    this.fs.symlinkSync(this.nm(src), this.nm(dst), to_dir ? 'dir' : 'file');
+    return 0;
+  } catch (err) {
+    this.raise_nodejs_error(err, raise_unix);
+  }
+}
+MlNodeDevice.prototype.readlink = function(name, raise_unix) {
+  try {
+    var js_stats = this.fs.readlinkSync(this.nm(name), 'utf8');
+    return caml_string_of_jsbytes(js_stats);
```

Comment:
- please rename js_stats.
- do we want to call caml_string_of_jsbytes or caml_string_of_jsstring ?

Reply:
I've renamed the variable. I'm actually still unclear about the difference between `caml_string_of_jsbytes` and `caml_string_of_jsstring`? If I change this to `caml_string_of_jsstring`, do I also need to change the 2 references inside `raise_nodejs_error`?

Reply:
I'm noticing that `caml_string_of_jsstring` has the `//If: js-string` requirement. What are the implications of that if this uses `caml_string_of_jsstring`?

We aren't using the `js-string` option in Grain so I'm unfamiliar.

Reply:
```//If: js-string``` is to have conditional stubs based on the js-string option see https://github.com/ocsigen/js_of_ocaml/pull/923. In short, one can choose the memory representation of ocaml strings.

`caml_string_of_jsbyte` should be used when the javascript string represent sequence of bytes, where all chars fall into the 0..255 range. `caml_string_of_jsstring` should be used for normal javascript strings, utf16 encoded. The utf8 encoding  argument is confusing here but I think you still get back a regular javascript string, utf16 encoded. I would use caml_string_of_jsstring and update the two other references.